src/System.Drawing.Primitives/src/System/Drawing/Color.cs

```diff
@@ -383,7 +383,8 @@ public string Name
             {
                 if ((state & StateNameValid) != 0)
```

Comment:
@safern I lack the domain logic here, is it correct to assume when condition `(state & StateNameValid) != 0` is true then name is expected not to be null? 

I assume this is a correct assumption and that is why I added the Debug.Assert here.

Reply:
I believe that's valid, since the only place Color is constructed with StateNameValid is:
https://github.com/dotnet/corefx/blob/9e056bb00f9e41de68579e8b27ed1f9470d89ad7/src/System.Drawing.Primitives/src/System/Drawing/Color.cs#L473
and that's just after `name` was used to try to index into a dictionary; if `name` was `null`, that `TryGetValue` call would throw an `ArgumentNullException`.